src/main/haskell/kore/src/Kore/SMT/SMT.hs

```diff
@@ -225,9 +230,11 @@ patternToSMT sloppy p =
             => CommonPurePattern Object
             -> String
             -> Translating (SBV a)
-        goLiteral (DV_ s (BuiltinDomainPattern (StringLiteral_ i))) hookName
+        goLiteral p@(DV_ s (BuiltinDomainPattern (StringLiteral_ i))) hookName
          | (getHookString $ getSortHook s) == hookName
-             = return $ literal $ read i
+             = liftEither $ note
+                 (MalformedDVLiteral p)
+                 (literal <$> readMaybe i)
```

Comment:
The domain value literals should be parsed using the appropriate parser in `Kore.Builtin.Int` or `Kore.Builtin.Bool`. I think @u-- is taking care of this.

In the long-term, all these builtins will be decoded into an `Integer` or `Bool` before they ever get to SMT.

Reply:
The only reason for this change is to make the integration tests work again, for @traiansf 's PR. I would rather not do too big of a change, especially since @u-- has more changes coming up. This change was suggested by him, actually.